Leftover doctest files cause Pytest errors in master

<!-- This comments are hidden when you submit the issue,
so you do not need to remove them! -->

<!-- Please be sure to check out our contributing guidelines,
https://github.com/astropy/astropy/blob/master/CONTRIBUTING.md .
Please be sure to check out our code of conduct,
https://github.com/astropy/astropy/blob/master/CODE_OF_CONDUCT.md . -->

<!-- Please have a search on our GitHub repository to see if a similar
issue has already been posted.
If a similar issue is closed, have a quick look to see if you are satisfied
by the resolution.
If not please go ahead and open an issue! -->

<!-- Please check that the development version still produces the same bug.
You can install development version with
pip install git+https://github.com/astropy/astropy
command. -->

### Description
Running pytest causes doctests to leave lots of test files lying around in the root directory that are not cleaned up. When running pytest a second time, some doctests fail because they do not expect the files to exist.

This issue was supposed to be addressed in the workaround in #9439. But I'm getting it again now with master, and figured it was worth an issue since it's causing tests to fail.

### Expected behavior
Pytest should run without errors no matter how many times you run it.

### Actual behavior
There are 6 doctests that fail if pytest was previously run without cleaning untracked files from the working directory..

### Steps to Reproduce
<!-- Ideally a code example could be provided so we can run it ourselves. -->
<!-- If you are pasting code, use triple backticks (```) around
your code snippet. -->
<!-- If necessary, sanitize your screen output to be pasted so you do not
reveal secrets like tokens and passwords. -->

1. Check for files: compressed_image.fits, test_file.fits, my_image.fits, variable_length_table.fits, newtable.fits, newtable2.fits, new.fits, new1.fits, new2.fits out.fits, table1.fits, table2.fits, table3.fits, values.dat. Remove them if they exist.
2. Run `pytest --doctest-ignore-import-errors`. It should complete without errors
3. Run `pytest --doctest-ignore-import-errors`. Six of the tests will fail. The failure output is below:
```
=================================== FAILURES ===================================
_____________________________ [doctest] index.rst ______________________________
631
632 Next, we create a :class:`PrimaryHDU` object to encapsulate the data::
633
634     >>> hdu = fits.PrimaryHDU(n)
635
636 We then create an :class:`HDUList` to contain the newly created primary HDU, and write to
637 a new file::
638
639     >>> hdul = fits.HDUList([hdu])
640     >>> hdul.writeto('new1.fits')
UNEXPECTED EXCEPTION: OSError("File 'new1.fits' already exists.")
Traceback (most recent call last):
  File "/Users/gsimonian/miniconda3/envs/astropy-dev/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest index.rst[60]>", line 1, in <module>
  File "/Users/gsimonian/dev/astropy/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/hdu/hdulist.py", line 933, in writeto
    fileobj = _File(fileobj, mode=mode, overwrite=overwrite)
  File "/Users/gsimonian/dev/astropy/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 175, in __init__
    self._open_filename(fileobj, mode, overwrite)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 553, in _open_filename
    self._overwrite_existing(overwrite, None, True)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 443, in _overwrite_existing
    raise OSError(f"File {self.name!r} already exists.")
OSError: File 'new1.fits' already exists.
/Users/gsimonian/dev/astropy/docs/io/fits/index.rst:640: UnexpectedException
_____________________________ [doctest] image.rst ______________________________
206 an example of what happens to the :attr:`~ImageHDU.data` attribute after the
207 :meth:`~ImageHDU.scale` call::
208
209     >>> hdu = fits.PrimaryHDU(np.array([0., 1, 2, 3]))
210     >>> print(hdu.data)  # doctest: +FLOAT_CMP
211     [0. 1. 2. 3.]
212     >>> hdu.scale('int16', bzero=32768)
213     >>> print(hdu.data)  # now the data has storage values
214     [-32768 -32767 -32766 -32765]
215     >>> hdu.writeto('new.fits')
UNEXPECTED EXCEPTION: OSError("File 'new.fits' already exists.")
Traceback (most recent call last):
  File "/Users/gsimonian/miniconda3/envs/astropy-dev/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest image.rst[28]>", line 1, in <module>
  File "/Users/gsimonian/dev/astropy/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/hdu/base.py", line 371, in writeto
    hdulist.writeto(name, output_verify, overwrite=overwrite,
  File "/Users/gsimonian/dev/astropy/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/hdu/hdulist.py", line 933, in writeto
    fileobj = _File(fileobj, mode=mode, overwrite=overwrite)
  File "/Users/gsimonian/dev/astropy/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 175, in __init__
    self._open_filename(fileobj, mode, overwrite)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 553, in _open_filename
    self._overwrite_existing(overwrite, None, True)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 443, in _overwrite_existing
    raise OSError(f"File {self.name!r} already exists.")
OSError: File 'new.fits' already exists.
/Users/gsimonian/dev/astropy/docs/io/fits/usage/image.rst:215: UnexpectedException
_____________________________ [doctest] table.rst ______________________________
138
139 ..
140   EXAMPLE START
141   Selecting Records in a Table Using a "Mask Array"
142
143 Assuming the table's second field as having the name 'magnitude', an output
144 table containing all the records of magnitude > -0.5 from the input table is
145 generated::
146
147     >>> with fits.open(fits_table_filename) as hdul:
UNEXPECTED EXCEPTION: OSError("File 'newtable.fits' already exists.")
Traceback (most recent call last):
  File "/Users/gsimonian/miniconda3/envs/astropy-dev/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest table.rst[13]>", line 6, in <module>
  File "/Users/gsimonian/dev/astropy/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/hdu/base.py", line 1582, in writeto
    hdulist.writeto(name, output_verify, overwrite=overwrite,
  File "/Users/gsimonian/dev/astropy/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/hdu/hdulist.py", line 933, in writeto
    fileobj = _File(fileobj, mode=mode, overwrite=overwrite)
  File "/Users/gsimonian/dev/astropy/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 175, in __init__
    self._open_filename(fileobj, mode, overwrite)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 553, in _open_filename
    self._overwrite_existing(overwrite, None, True)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 443, in _overwrite_existing
    raise OSError(f"File {self.name!r} already exists.")
OSError: File 'newtable.fits' already exists.
/Users/gsimonian/dev/astropy/docs/io/fits/usage/table.rst:147: UnexpectedException
___________________________ [doctest] unfamiliar.rst ___________________________
190     >>> col1 = fits.Column(
191     ...    name='var', format='PI()',
192     ...    array=np.array([[45, 56], [11, 12, 13]], dtype=np.object_))
193     >>> col2 = fits.Column(name='xyz', format='2I', array=[[11, 3], [12, 4]])
194     >>> hdu = fits.BinTableHDU.from_columns([col1, col2])
195     >>> data = hdu.data
196     >>> data  # doctest: +SKIP
197     FITS_rec([([45, 56], [11,  3]), ([11, 12, 13], [12,  4])],
198              dtype=(numpy.record, [('var', '<i4', (2,)), ('xyz', '<i2', (2,))]))
199     >>> hdu.writeto('variable_length_table.fits')
UNEXPECTED EXCEPTION: OSError("File 'variable_length_table.fits' already exists.")
Traceback (most recent call last):
  File "/Users/gsimonian/miniconda3/envs/astropy-dev/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest unfamiliar.rst[25]>", line 1, in <module>
  File "/Users/gsimonian/dev/astropy/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/hdu/base.py", line 1582, in writeto
    hdulist.writeto(name, output_verify, overwrite=overwrite,
  File "/Users/gsimonian/dev/astropy/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/hdu/hdulist.py", line 933, in writeto
    fileobj = _File(fileobj, mode=mode, overwrite=overwrite)
  File "/Users/gsimonian/dev/astropy/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 175, in __init__
    self._open_filename(fileobj, mode, overwrite)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 553, in _open_filename
    self._overwrite_existing(overwrite, None, True)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 443, in _overwrite_existing
    raise OSError(f"File {self.name!r} already exists.")
OSError: File 'variable_length_table.fits' already exists.
/Users/gsimonian/dev/astropy/docs/io/fits/usage/unfamiliar.rst:199: UnexpectedException
____________________________ [doctest] ccddata.rst _____________________________
098            fill_value=999999)
099
100 A method for converting a `~astropy.nddata.CCDData` object to a FITS HDU list
101 is also available. It converts the metadata to a FITS header:
102
103     >>> hdulist = ccd_masked.to_hdu()
104
105 You can also write directly to a FITS file:
106
107     >>> ccd_masked.write('my_image.fits')
UNEXPECTED EXCEPTION: OSError("File 'my_image.fits' already exists.")
Traceback (most recent call last):
  File "/Users/gsimonian/miniconda3/envs/astropy-dev/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest ccddata.rst[19]>", line 1, in <module>
  File "/Users/gsimonian/dev/astropy/astropy/nddata/mixins/ndio.py", line 101, in __call__
    registry.write(self._instance, *args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/io/registry.py", line 563, in write
    writer(data, *args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/nddata/ccddata.py", line 714, in fits_ccddata_writer
    hdu.writeto(filename, **kwd)
  File "/Users/gsimonian/dev/astropy/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/hdu/hdulist.py", line 933, in writeto
    fileobj = _File(fileobj, mode=mode, overwrite=overwrite)
  File "/Users/gsimonian/dev/astropy/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 175, in __init__
    self._open_filename(fileobj, mode, overwrite)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 553, in _open_filename
    self._overwrite_existing(overwrite, None, True)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 443, in _overwrite_existing
    raise OSError(f"File {self.name!r} already exists.")
OSError: File 'my_image.fits' already exists.
/Users/gsimonian/dev/astropy/docs/nddata/ccddata.rst:107: UnexpectedException
_____________________________ [doctest] index.rst ______________________________
234
235 ..
236   EXAMPLE END
237
238 Reading and Writing
239 -------------------
240
241 A `~astropy.nddata.CCDData` can be saved to a FITS file::
242
243     >>> ccd.write('test_file.fits')
UNEXPECTED EXCEPTION: OSError("File 'test_file.fits' already exists.")
Traceback (most recent call last):
  File "/Users/gsimonian/miniconda3/envs/astropy-dev/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest index.rst[47]>", line 1, in <module>
  File "/Users/gsimonian/dev/astropy/astropy/nddata/mixins/ndio.py", line 101, in __call__
    registry.write(self._instance, *args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/io/registry.py", line 563, in write
    writer(data, *args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/nddata/ccddata.py", line 714, in fits_ccddata_writer
    hdu.writeto(filename, **kwd)
  File "/Users/gsimonian/dev/astropy/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/hdu/hdulist.py", line 933, in writeto
    fileobj = _File(fileobj, mode=mode, overwrite=overwrite)
  File "/Users/gsimonian/dev/astropy/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 175, in __init__
    self._open_filename(fileobj, mode, overwrite)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 553, in _open_filename
    self._overwrite_existing(overwrite, None, True)
  File "/Users/gsimonian/dev/astropy/astropy/io/fits/file.py", line 443, in _overwrite_existing
    raise OSError(f"File {self.name!r} already exists.")
OSError: File 'test_file.fits' already exists.
/Users/gsimonian/dev/astropy/docs/nddata/index.rst:243: UnexpectedException
------------------------------ Captured log call -------------------------------
INFO     astropy:ccddata.py:265 array provided for uncertainty; assuming it is a StdDevUncertainty.
=========================== short test summary info ============================
FAILED docs/io/fits/index.rst::index.rst
FAILED docs/io/fits/usage/image.rst::image.rst
FAILED docs/io/fits/usage/table.rst::table.rst
FAILED docs/io/fits/usage/unfamiliar.rst::unfamiliar.rst
FAILED docs/nddata/ccddata.rst::ccddata.rst
FAILED docs/nddata/index.rst::index.rst
==== 6 failed, 13298 passed, 1180 skipped, 61 xfailed in 360.37s (0:06:00) =====
```
4. If you delete the above files, pytest runs without failures again.

### System Details
<!-- Even if you do not think this is necessary, it is useful information for the maintainers.
Please run the following snippet and paste the output below:
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import numpy; print("Numpy", numpy.__version__)
import astropy; print("astropy", astropy.__version__)
import scipy; print("Scipy", scipy.__version__)
import matplotlib; print("Matplotlib", matplotlib.__version__)
-->

Python 3.8.3 (default, May 19 2020, 13:54:14)
[Clang 10.0.0 ] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
import platform; print(platform.platform())
macOS-10.15.4-x86_64-i386-64bit
import sys; print("Python", sys.version)
Python 3.8.3 (default, May 19 2020, 13:54:14)
[Clang 10.0.0 ]
import numpy; print("Numpy", numpy.__version__)
Numpy 1.18.1
import astropy; print("astropy", astropy.__version__)
astropy 4.2.dev189+g305a10652
import scipy; print("Scipy", scipy.__version__)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'scipy'
import matplotlib; print("Matplotlib", matplotlib.__version__)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'matplotlib'

--- a/docs/conftest.py
+++ b/docs/conftest.py
@@ -29,7 +29,7 @@ def _docdir(request):
     """Run doctests in isolated tmpdir so outputs do not end up in repo"""
     # Trigger ONLY for doctestplus
     doctest_plugin = request.config.pluginmanager.getplugin("doctestplus")
-    if isinstance(request.node, doctest_plugin._doctest_textfile_item_cls):
+    if isinstance(request.node.parent, doctest_plugin._doctest_textfile_item_cls):
         # Don't apply this fixture to io.rst.  It reads files and doesn't write
         if "io.rst" not in request.node.name:
             tmpdir = request.getfixturevalue('tmpdir')

